### PR TITLE
Add "Warnings" section to doc/devguide.md

### PR DESCRIPTION
**Pull request by onalante-msft**
A recent PR of mine caused a pipeline break due to the backported PR being merged before the main PR. The breakage was due to the pipelines for the main branch utilizing images built from 1.2, something which I was not aware of. This PR adds a "Warnings" section to the developer guide initially starting with this note. Similar unexpected failure modes can be added as they are found by others.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [ ] local testing done.

### CODE REVIEW

File: doc/devguide.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ The C# development setup is described below. The Rust development setup is descr
 
 If you want to run tests outside of the pipelines, you will need to be running linux.
 
+## Warnings
+
+- If you have upgraded the most recent API version and have backported to 1.2, make sure PR for main is merged before the PR to 1.2. This is because the pipelines for main rely on images built from 1.2.
```
Comment (damonbarry) [inline doc/devguide.md:10]: > the pipelines for main rely on images built from 1.2

I don't think that's true. Was the pipeline manually triggered? Or was this an automated run?

Comment (onalante-msft) [reply doc/devguide.md:10]: It was an automated run, and I was informed by someone internal that this was occurring.

Comment (damonbarry) [reply doc/devguide.md:10]: @onalante-msft Can you dm a link to the pipeline?

Comment (onalante-msft) [reply doc/devguide.md:10]: Okay, it looks like we have internal folks that use edgelet from main but images from 1.2, hence the failure. I will revise the document to indicate this.

Comment (damonbarry) [reply doc/devguide.md]: If we have internal folks doing stuff with internal pipelines, maybe we can just document this internally to avoid confusion....

Comment (onalante-msft) [reply doc/devguide.md]: Yes, that sounds like a better plan. I was originally unsure of where this issue arose. I will close this PR.

**Pull request closed without merging**